### DISCUSSION TITLE
Make output filenames changeable

### DISCUSSION THREAD

**Discussion by Kimvana** [lacourjansenlab/NISE_2017#16] [Team Posts]
In my experience, the fact that NISE always has the same output filenames is frustrating to work with - you cannot simply call multiple input files for different calculations, they would overwrite each others files. My suggestion: add (a) keyword(s) that allows for changing the names.

Option 1:
A single keyword. The submitted text is inserted into the filename. So, if someone would say '[keyword] life1000', NISE would not use the file NISE.log, but NISE_life1000.log instead.
Advantage: only a single keyword to implement, and a single one for users to change, that changes all files created by NISE. 
Disadvantage: no control over individual files.

Option 2:
Create multiple keywords. Considering how 2DFFT alone creates 12 files, probably not one for each file, perhaps one for each category? 
Advantage: more control, could even allow removing the default part (so 'mycoolcalculation.log' becomes available), option to only do this to part of the parameters
Disatvantage: easier to accidentaly forget one, harder to implement.

**Comment by GHlacour** [comment]
Should be doable and a good idea. My first inclination is option 1. One point of flexibility is that the 2DFFT code has a a separate input reader which may be utilised for the many output files that it creates.
Maybe "FileLabel" would be a good keyword.

**Comment by GHlacour** [comment]
The NISE.log file is also always the same. Could be good to have technique dependent log file names automatically.
